src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ExceptionTest/ExceptionTest.cs

```diff
@@ -201,11 +201,20 @@ public static void ExceptionTests()
         }
 
         // Synapse: 110003;Invalid user or password
-        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup), nameof(DataTestUtility.IsNotAzureServer))]
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
         public static void VariousExceptionTests()
         {
             SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString);
-
+            if (!DataTestUtility.IsNotAzureServer())
+            {
+                // Strip the password in connection string if Authentication=Active Directory Managed Identity as it can not be used with a Password
+                if (builder.Authentication == SqlAuthenticationMethod.ActiveDirectoryManagedIdentity)
+                {
+                    string[] removeKeys = { "Password", "PWD" };
+                    string connStr = DataTestUtility.RemoveKeysInConnStr(DataTestUtility.TCPConnectionString, removeKeys);
+                    builder = new SqlConnectionStringBuilder(connStr);
+                }
+            }
```

Comment:
Only verifying the auth is sufficient. 
```suggestion
            // Strip the password in connection string if Authentication=Active Directory Managed Identity as it can not be used with a Password
            if (builder.Authentication == SqlAuthenticationMethod.ActiveDirectoryManagedIdentity)
            {
                string[] removeKeys = { "Password", "PWD" };
                string connStr = DataTestUtility.RemoveKeysInConnStr(DataTestUtility.TCPConnectionString, removeKeys);
                builder = new SqlConnectionStringBuilder(connStr);
            }
```